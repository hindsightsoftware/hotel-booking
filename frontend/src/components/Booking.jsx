import React from 'react';
import request from 'superagent';

export default class Booking extends React.Component {

    constructor() {
        super();
        this.state = { 
            booking : {
                "firstname": null,
                "lastname": null,
                "totalprice": null,
                "depositpaid": null,
                "additionalneeds": null,
                "bookingdates": {
                    "checkin": null,
                    "checkout": null
                }
            }
        };

        this.destroyMe = this.destroyMe.bind(this);
    }

    componentDidMount() {
        if (process.env.NODE_ENV === 'test') return
        
        request
            .get('/booking/' + this.props.id)
            .then((res, err) => {
              const booking = res.body;
              this.setState({booking});
            }).catch(e => {
                console.log(e);
            })
    }

    destroyMe() {
        request
            .delete('/booking/' + this.props.id)
            .auth('admin', 'password123')
            .then((res, err) => {
                window.location.reload();
            }).catch(e => {
                console.log(e);
            })
    }

    render() {
        return (
            <div className="row">
                <div className="col-md-1">{this.state.booking.firstname}</div>
                <div className="col-md-2">{this.state.booking.lastname}</div>
                <div className="col-md-1">{this.state.booking.totalprice}</div>
                <div className="col-md-1">{String(this.state.booking.depositpaid)}</div>
                <div className="col-md-2">{this.state.booking.additionalneeds}</div>
                <div className="col-md-2">{this.state.booking.bookingdates.checkin}</div>
                <div className="col-md-2">{this.state.booking.bookingdates.checkin}</div>
                <div className="col-md-1"><i className="glyphicon glyphicon-edit"></i> 
                    <button onClick={this.destroyMe} className="delete"><i className="glyphicon glyphicon-remove"></i></button>
                </div>      
            </div>
        );
      }

}