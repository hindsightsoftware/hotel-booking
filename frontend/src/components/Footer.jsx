import React from 'react';

export default class Footer extends React.Component {
  render() {
    return (
        <div id="footer">
            <div className="container">
                <p className="text-muted credit">Restful-booker created by <a href="http://mwtestconsultancy.co.uk">Mark Winteringham</a></p>
            </div>
        </div>);
  }
}