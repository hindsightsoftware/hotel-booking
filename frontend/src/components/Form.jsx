import React from 'react';
import request from 'superagent';

export default class Form extends React.Component {

    constructor(props){
        super(props);

        this.updateValue = this.updateValue.bind(this);
    }

    updateValue(){
        let bookingModel = {
            firstname : this.firstNameValue.value,
            lastname : this.lastNameValue.value,
            totalprice : this.priceValue.value,
            depositpaid : this.depositPaidValue.value,
            bookingdates : {
                checkin : this.checkInValue.value,
                checkout : this.checkOutValue.value
            },
            additionalneeds : this.additionalNeedsValue.value
        }

        request
            .post('/booking')
            .set('Content-Type', 'application/json')
            .send(bookingModel)
            .then((res, err) => {
                window.location.reload();
            }).catch(e => {
                console.log(e);
            })
    }

    render() {
        return (
            <div className="row form">
                <div className="col-md-1"><input ref={el => this.firstNameValue=el} type="text" id="firstname" className="form-control input-sm" placeholder="Firstname" /></div>
                <div className="col-md-2"><input ref={el => this.lastNameValue=el} type="text" id="lastname" className="form-control input-sm" placeholder="Lastname" /></div>
                <div className="col-md-1"><input ref={el => this.priceValue=el} type="text" id="totalprice" className="form-control input-sm" placeholder="Price" /></div>
                <div className="col-md-1"><input ref={el => this.depositPaidValue=el} type="text" id="depositpaid" className="form-control input-sm" placeholder="Deposit" /></div>
                <div className="col-md-2"><input ref={el => this.additionalNeedsValue=el} type="text" id="additionalneeds" className="form-control input-sm" placeholder="Additional" /></div>
                <div className="col-md-2"><input ref={el => this.checkInValue=el} type="text" id="checkin" className="form-control input-sm" placeholder="Checkin" /></div>
                <div className="col-md-2"><input ref={el => this.checkOutValue=el} type="text" id="checkout" className="form-control input-sm" placeholder="Checkout" /></div>
                <div className="col-md-1"><button onClick={this.updateValue} className="btn btn-sm addButton">Add</button></div>
            </div>
        )
    }

}