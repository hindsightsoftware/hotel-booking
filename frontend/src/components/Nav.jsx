import React from 'react';

export default class Nav extends React.Component {
  render() {
    return (
        <div className="navbar navbar-default navbar-fixed-top">
        <div className="container">
          <div className="navbar-header">
            <button type="button" className="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
              <span className="icon-bar"></span>
              <span className="icon-bar"></span>
              <span className="icon-bar"></span>
            </button>
            <a className="navbar-brand" href="#">Restful booker</a>
          </div>
        </div>
      </div>);
  }
}