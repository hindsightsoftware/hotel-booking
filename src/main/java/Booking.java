import java.util.Date;

public class Booking {

    private String firstname;
    private String lastname;
    private int price;
    private boolean booked;
    private BookingDates bookingDates;
    private String additional;

    public Booking(String firstname, String lastname, int price, boolean booked, Date checkin, Date checkout, String additional) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.price = price;
        this.booked = booked;
        this.bookingDates = new BookingDates(checkin, checkout);
        this.additional = additional;
    }

    @Override
    public String toString() {
        return "Booking{" +
                "firstname='" + firstname + '\'' +
                "\n, lastname='" + lastname + '\'' +
                "\n, price=" + price +
                "\n, booked=" + booked +
                "\n, bookingDates=" + bookingDates.toString() +
                "\n, additional='" + additional + '\'' +
                "\n}";
    }
}
