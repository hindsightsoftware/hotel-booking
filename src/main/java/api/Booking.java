package api;

import model.BookingModel;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.Date;

@RestController
public class Booking {

    @RequestMapping(value = "/booking", method = RequestMethod.GET)
    public BookingModel returnBooking() {
        Date checkin = new Date();
        checkin.setTime(1514764800);
        Date checkout = new Date();
        checkout.setTime(1514851200);

        BookingModel booking = new BookingModel("Mark", "Winteringham", 111, true, checkin, checkout, "Breakfast");

        return booking;
    }

}
