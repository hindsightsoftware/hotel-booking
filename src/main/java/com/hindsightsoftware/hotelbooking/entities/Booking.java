package com.hindsightsoftware.hotelbooking.entities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
public class Booking {
    @Id
    @GeneratedValue(strategy=GenerationType.AUTO)
    private Long id;
    private String firstName;
    private String lastName;
    private int totalPrice;
    private boolean depositPaid;
    private Date checkIn;
    private Date checkOut;
    private String additionalNeeds;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(int totalPrice) {
        this.totalPrice = totalPrice;
    }

    public boolean isDepositPaid() {
        return depositPaid;
    }

    public void setDepositPaid(boolean depositPaid) {
        this.depositPaid = depositPaid;
    }

    public Date getCheckIn() {
        return checkIn;
    }

    public void setCheckIn(Date checkIn) {
        this.checkIn = checkIn;
    }

    public Date getCheckOut() {
        return checkOut;
    }

    public void setCheckOut(Date checkOut) {
        this.checkOut = checkOut;
    }

    public String getAdditionalNeeds() {
        return additionalNeeds;
    }

    public void setAdditionalNeeds(String additionalNeeds) {
        this.additionalNeeds = additionalNeeds;
    }
}
