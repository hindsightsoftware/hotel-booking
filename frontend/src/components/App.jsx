import React from 'react';
import request from 'superagent';
import Booking from './Booking.jsx';
import Nav from './Nav.jsx';
import Header from './Header.jsx';
import Form from './Form.jsx';

export default class App extends React.Component {
    constructor() {
        super();
        this.state = { bookings : [] };
    }

  componentDidMount() {
    if (process.env.NODE_ENV === 'test') return // https://github.com/facebookincubator/create-react-app/issues/3482 - Allegedly will be fixed soon

    request
        .get('/booking')
        .then((res, err) => {
            const bookings = res.body;
            this.setState({bookings});
        }).catch(e => {
            console.log(e);
        })
  }

  render() {
    let bookings = null;
    if(this.state.bookings.length > 0){
        bookings = this.state.bookings.map((name, index) => {
            return <div key={name.id}><Booking id={name.id} /></div>
        })
    } else {
        bookings = <p>No bookings found</p>
    }

    return (
        <div id="wrap">
            <Nav />
            <div className="container">
                <Header />
                <div id="bookings">
                    {bookings}
                </div>
                <Form />
            </div>
        </div>);
  }
}