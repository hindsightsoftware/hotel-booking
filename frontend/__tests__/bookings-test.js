import React from 'react';
import App from '../src/components/App.jsx';
import Booking from '../src/components/Booking.jsx';
import Nav from '../src/components/Nav.jsx';
import Footer from '../src/components/Footer.jsx';
import Header from '../src/components/Header.jsx';
import Form from '../src/components/Form.jsx';
import nock from 'nock';

test('App component', () => {
    const testBookings = [
        {
            "id": 1
        },{
            "id": 2
        }
    ]

    const appWrapper = shallow(
        <App />
    );

    appWrapper.setState({bookings : testBookings});

    expect(appWrapper).toMatchSnapshot();
});

test('Booking component', () => {
    const testBooking = {
        "firstname": "Billy",
        "lastname": "Brown",
        "totalprice": 90,
        "depositpaid": true,
        "additionalneeds": "Jam",
        "bookingdates": {
            "checkin": "2018-01-01",
            "checkout": "2019-01-01"
        }
    }

    const bookingEntry = shallow(
        <Booking id="1" />
    )

    bookingEntry.setState({ booking: testBooking })

    expect(bookingEntry).toMatchSnapshot();
});

test('Nav component', () => {
    const navBar = shallow(
        <Nav />
    )

    expect(navBar).toMatchSnapshot();
});

test('Footer component', () => {
    const footer = shallow(
        <Footer />
    )

    expect(footer).toMatchSnapshot();
});

test('Header component', () => {
    const header = shallow(
        <Header />
    )

    expect(header).toMatchSnapshot();
});

test('Form component', () => {
    const form = shallow(
        <Form />
    )

    expect(form).toMatchSnapshot();
})

test('Submit booking', (done) => {
    
    const expectedPayload = { 
        "firstname" : "Mark",
        "lastname" : "Winteringham",
        "totalprice" : "123",
        "depositpaid" : "true",
        "bookingdates" : {
            "checkin" : "2018-01-01",
            "checkout" : "2018-01-02"
        }, "additionalneeds" : "Breakfast"
    }

    nock('http://localhost')
      .post('/booking', expectedPayload)
      .reply(201, () => {
          done();
      })

    const form = mount(
        <Form />
    )

    form.find('#firstname').instance().value = "Mark"
    form.find('#lastname').instance().value = "Winteringham"
    form.find('#totalprice').instance().value = "123"
    form.find('#depositpaid').instance().value = "true"
    form.find('#additionalneeds').instance().value = "Breakfast"
    form.find('#checkin').instance().value = "2018-01-01"
    form.find('#checkout').instance().value = "2018-01-02"
    form.find('button').simulate('click');
})