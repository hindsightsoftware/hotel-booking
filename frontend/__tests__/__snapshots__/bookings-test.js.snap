// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<div
  id="wrap"
>
  <Nav />
  <div
    className="container"
  >
    <Header />
    <div
      key="1"
    >
      <Booking
        id={1}
      />
    </div>
    <div
      key="2"
    >
      <Booking
        id={2}
      />
    </div>
    <Form />
  </div>
</div>
`;

exports[`Booking component 1`] = `
<div
  className="row"
>
  <div
    className="col-md-1"
  >
    Billy
  </div>
  <div
    className="col-md-2"
  >
    Brown
  </div>
  <div
    className="col-md-1"
  >
    90
  </div>
  <div
    className="col-md-1"
  >
    true
  </div>
  <div
    className="col-md-2"
  >
    Jam
  </div>
  <div
    className="col-md-2"
  >
    2018-01-01
  </div>
  <div
    className="col-md-2"
  >
    2018-01-01
  </div>
  <div
    className="col-md-1"
  >
    <i
      className="glyphicon glyphicon-edit"
    />
    <button
      className="delete"
      onClick={[Function]}
    >
      <i
        className="glyphicon glyphicon-remove"
      />
    </button>
  </div>
</div>
`;

exports[`Form component 1`] = `
<div
  className="row"
>
  <div
    className="col-md-1"
  >
    <input
      className="form-control input-sm"
      id="firstname"
      placeholder="Firstname"
      type="text"
    />
  </div>
  <div
    className="col-md-2"
  >
    <input
      className="form-control input-sm"
      id="lastname"
      placeholder="Lastname"
      type="text"
    />
  </div>
  <div
    className="col-md-1"
  >
    <input
      className="form-control input-sm"
      id="totalprice"
      placeholder="Price"
      type="text"
    />
  </div>
  <div
    className="col-md-1"
  >
    <input
      className="form-control input-sm"
      id="depositpaid"
      placeholder="Deposit?"
      type="text"
    />
  </div>
  <div
    className="col-md-2"
  >
    <input
      className="form-control input-sm"
      id="additionalneeds"
      placeholder="Additional"
      type="text"
    />
  </div>
  <div
    className="col-md-2"
  >
    <input
      className="form-control input-sm"
      id="checkin"
      placeholder="Checkin"
      type="text"
    />
  </div>
  <div
    className="col-md-2"
  >
    <input
      className="form-control input-sm"
      id="checkout"
      placeholder="Checkout"
      type="text"
    />
  </div>
  <div
    className="col-md-1"
  >
    <button
      className="btn btn-sm"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;

exports[`Header component 1`] = `
<div
  className="row"
>
  <div
    className="col-md-1"
  >
    <p
      className="lead"
    >
      Firstname
    </p>
  </div>
  <div
    className="col-md-2"
  >
    <p
      className="lead"
    >
      Lastname
    </p>
  </div>
  <div
    className="col-md-1"
  >
    <p
      className="lead"
    >
      Price
    </p>
  </div>
  <div
    className="col-md-1"
  >
    <p
      className="lead"
    >
      Deposit
    </p>
  </div>
  <div
    className="col-md-2"
  >
    <p
      className="lead"
    >
      Additional
    </p>
  </div>
  <div
    className="col-md-2"
  >
    <p
      className="lead"
    >
      Checkin
    </p>
  </div>
  <div
    className="col-md-2"
  >
    <p
      className="lead"
    >
      Checkout
    </p>
  </div>
  <div
    className="col-md-1"
  />
</div>
`;

exports[`Nav component 1`] = `
<div
  className="navbar navbar-default navbar-fixed-top"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-toggle"
        data-target=".navbar-collapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
      <a
        className="navbar-brand"
        href="#"
      >
        Restful booker
      </a>
    </div>
  </div>
</div>
`;
